src/utils.ts

```diff
@@ -0,0 +1,74 @@
+import * as fs from "fs";
+import * as core from "@actions/core";
+import * as ini from "ini";
+import * as path from "path";
+import * as os from "os";
+
+export function checkStorageDriver(filePaths: string[]): void {
+    const filePath = checkStorageFile(filePaths);
+    if (filePath) {
+        core.info("Reading storage file to find out the storage driver used");
+        const fileContent = ini.parse(fs.readFileSync(filePath, "utf-8"));
+        const storageDriver = fileContent.storage.driver;
+        core.info(`Storage driver is set to "${storageDriver}"`);
+
+        // Check for storage-driver, if overlay then and
+        // add mount program as "fuse-overlayfs" in ~/.config/containers/storage.conf
+        if (storageDriver === "overlay") {
+            // Backup file if ~/.config/containers/storage.conf exists
+            if (filePath === filePaths[0]) {
+                backupFile(filePath);
+            }
+            addMountProgram();
+        }
+    }
+    else {
+        addMountProgram();
+    }
+}
+
+function checkStorageFile(filePaths: string[]): string {
+    for (const filePath of filePaths) {
+        core.debug(`Checking if the storage file exists at ${filePath}`);
+        if (fs.existsSync(filePath)) {
+            core.debug(`Storage file exists at ${filePath}`);
+            return filePath;
+        }
+    }
+    return "";
+}
+
+function backupFile(filePath: string): void {
+    core.debug(`Creating backup of the existing "storage.conf" file`);
+    fs.copyFileSync(`${filePath}`, `${filePath}.backup`);
```

Comment:
put .backup in a constant

or
```
function getBackupFileName(filePath: string): string {
    return filePath + ".backup";
}
```

and use that function everywhere you need it